deployment/cfn/application.py

```diff
@@ -0,0 +1,363 @@
+                               Value=GetAtt(app_server_lb,
+                                            'CanonicalHostedZoneNameID')))
+
+    def get_recent_application_server_ami(self):
+        try:
+            application_server_ami_id = self.get_input('AppServerAMI')
+        except MKUnresolvableInputError:
+            application_server_ami_id = get_recent_ami(self.aws_profile,
+                                                       'mmw-app-*',
+                                                       owner='self')
+
+        return application_server_ami_id
+
+    def create_security_groups(self):
+        app_server_lb_security_group_name = 'sgAppServerLoadBalancer'
+
+        app_server_lb_security_group = self.add_resource(ec2.SecurityGroup(
+            app_server_lb_security_group_name,
+            GroupDescription='Enables access to application servers via a '
+                             'load balancer',
+            VpcId=Ref(self.vpc_id),
+            SecurityGroupIngress=[
+                ec2.SecurityGroupRule(
+                    IpProtocol='tcp', CidrIp=ALLOW_ALL_CIDR, FromPort=p,
+                    ToPort=p
+                )
+                for p in [HTTP, HTTPS]
+            ],
+            SecurityGroupEgress=[
+                ec2.SecurityGroupRule(
+                    IpProtocol='tcp', CidrIp=VPC_CIDR, FromPort=p, ToPort=p
+                )
+                for p in [HTTP]
+            ],
+            Tags=self.get_tags(Name=app_server_lb_security_group_name)
+        ))
+
+        app_server_security_group_name = 'sgAppServer'
+
+        app_server_security_group = self.add_resource(ec2.SecurityGroup(
+            app_server_security_group_name,
+            GroupDescription='Enables access to application servers',
+            VpcId=Ref(self.vpc_id),
+            SecurityGroupIngress=[
+                ec2.SecurityGroupRule(
+                    IpProtocol='tcp', CidrIp=VPC_CIDR, FromPort=p, ToPort=p
+                )
+                for p in [SSH, HTTP]
+            ] + [
+                ec2.SecurityGroupRule(
+                    IpProtocol='tcp', SourceSecurityGroupId=Ref(sg),
+                    FromPort=HTTP, ToPort=HTTP
+                )
+                for sg in [app_server_lb_security_group]
+            ],
+            SecurityGroupEgress=[
+                ec2.SecurityGroupRule(
+                    IpProtocol='tcp', CidrIp=VPC_CIDR, FromPort=p, ToPort=p
+                )
+                for p in [GRAPHITE, POSTGRESQL, REDIS, STATSITE, RELP]
+            ] + [
+                ec2.SecurityGroupRule(
+                    IpProtocol='udp', CidrIp=VPC_CIDR, FromPort=p, ToPort=p
+                )
+                for p in [STATSITE]
+            ] + [
+                ec2.SecurityGroupRule(
+                    IpProtocol='tcp', CidrIp=ALLOW_ALL_CIDR, FromPort=p,
+                    ToPort=p
+                )
+                for p in [HTTP]
```

Comment:
If we're going to allow the app server full access to the outside on 80 we should probably add 443 since some things might be enforcing ssl everywhere.